components/button/floating-buttons.js

```diff
@@ -0,0 +1,260 @@
+				type: Object
+			},
+
+			_innerContainerStyle: {
+				type: Object
+			},
+
+			_buttonSpacerStyle: {
+				type: Object
+			}
+		};
+	}
+
+	static get styles() {
+		return [buttonStyles,
+			css`
+			:host {
+				box-sizing: border-box;
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-floating-buttons-container {
+				border-top: 1px solid transparent;
+				display: block;
+				margin: 0 auto;
+				width: 100%;
+			}
+			.d2l-floating-buttons-container.d2l-floating-buttons-floating {
+				animation: d2l-floating-buttons-animation 500ms ease-out;
+				-webkit-animation: d2l-floating-buttons-animation 500ms ease-out;
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+		this._containerClasses = {
+			'd2l-floating-buttons-container': true,
+			'd2l-floating-buttons-floating': false
+		};
+		this._innerContainerStyle = {
+			left: '',
+			right: '',
+			width: ''
+		};
+		this._buttonSpacerStyle = {
+			height: '',
+			display: ''
+		};
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('resize', () => {
```

Comment:
I think this can just be:
```
window.addEventListener('resize', this._reposition);
window.addEventListener('scroll', this._reposition);
```

Similarily, in disconnected callback, it can just be:
```
window.removeEventListener('resize', this._reposition);
window.removeEventListener('scroll', this._reposition);
```

Something important to note: the function object passed to `removeEventListener` should be the same instance as that provided to `addEventListener`. In the current code, I think they are two different instances, and the calls to `removeEventListener` would not properly remove the functions that were originally added.  Besides eliminating the unnecessary function wrapper, the suggested change addresses that issue as well.

Reply:
The problem that was causing is that when '_reposition' is called through that event `this` will be the window. this fixed that at them time. I'll double check, this perhaps that issues was unrelated.

Reply:
Arrow functions handle the `bind(this)` for you, which is why the wrapper function helped address that.  I think if `this._reposition = this._reposition.bind(this);` is moved to the constructor (per other comment), then `this` will be bound the way we want.

Reply:
you're correct that look much nicer. :D